src/core/public/theme/core_theme_provider.tsx

```diff
@@ -21,16 +23,26 @@ interface CoreThemeProviderProps {
   theme$: Observable<CoreTheme>;
 }
 
+const emotionCache = createCache({
+  key: 'eui-styles',
+  container: document.querySelector(`meta[name="${EUI_STYLES_GLOBAL}"]`) as HTMLElement,
+});
```

Comment:
Hum, shouldn't this also be used in the public version of the EuiProvider wrapper in `src/plugins/kibana_react/public/theme/kibana_theme_provider.tsx` ? (same for `modify`)

Reply:
Only the top-level theme provider needs the cache because it's the only provider that now also provides global styles. Those global styles only need to be included once and are the only styles that require the cache. Eventually, once EUI has converted all components to Emotion, the global style cache will be unnecessary and can be removed.

Reply:
Ok, Just to be sure we're on the same page, this `CoreThemeProvider` component is not used only once.

It is used as a 'top level' provider for chrome's react tree, but is also used for internal core's components, such as toasts, flyout and modal, as all those components are their own distinct react tree / ReactDOM.render call. 

Is it still fine?

Reply:
Yes, this is fine, thanks for clarifying. My main point was that a nested KibanaThemeProvider does not need to be aware of the cache.

I'll probably have a follow-up PR that ensures no duplication of the global styles from the various toasts, flyout and modal, etc. instances, but there will be no adverse affects in the meantime.